### DISCUSSION TITLE
Implementation of Scene Entity and Update to create rosbag

### DISCUSSION THREAD

**Discussion by lukashsmannheim** [foxglove/community#892] [General]
I am trying to visualize labels from the [view of delft dataset](https://github.com/tudelft-iv/view-of-delft-dataset) in foxglove. Here I am currently trying to implement the bounding boxes with the foxglove message package (specifically SceneUpdate, SceneEntity and CubePrimitve).

    for labelname in label_files[:num_msgs]:
    
    # generate absolute path of file
    label_file_path = label_path + labelname
    
    # get label data
    label_data = get_label_data(label_file_path)
    
    update = SceneUpdate()
    for row in label_data:
        
        entity = SceneEntity()
    
        # add header
        entity.timestamp = time
        entity.frame_id = os.path.splitext(labelname)[0]
        entity.lifetime = time_step
        entity.id = str(row['tracking-ID']) 
        entity.frame_locked = True
        
        cube = CubePrimitive()      
        
        # add position
        cube.pose.position.x = float(row['location_x'])
        cube.pose.position.y = float(row['location_y'])
        cube.pose.position.z = float(row['location_z'])
        
        # add size
        cube.size.x = float(row['dimensions_width'])
        cube.size.y = float(row['dimensions_length'])
        cube.size.z = float(row['dimensions_height'])
        
        # add orientations
        quaternions = (Rotation.from_euler('z', float(row['rotation']))).as_quat()
        cube.pose.orientation.x = quaternions[0]
        cube.pose.orientation.y = quaternions[1]
        cube.pose.orientation.z = quaternions[2]
        cube.pose.orientation.w = quaternions[3]
        
        # add cube to message
        entity.cubes.append(cube)
        
        update.entities.append(entity)
    
    label_msgs.append(update)
    timestamps.append(time)
    time = time + time_step

I am not fully sure if I am using the package correctly. After running following code to create the rosbag I am receiving a frame transform error in foxglove? 

![Screenshot from 2023-12-11 12-42-53](https://github.com/foxglove/community/assets/153269349/abd3a9d7-2ec4-4fbb-95f3-606614e3dab3)

How can I solve this issue? And why are transformations from frame to frame needed if my entities only last for one frame?

Thanks in advance!

**Comment by achim-k** [comment]
> ```entity.frame_id = os.path.splitext(labelname)[0]```

There is a misunderstanding, the frame here is supposed to be a coordinate frame (e.g. the camera frame) and not an image frame. The coordinate frame typically stays the same for all messages (image frames)

**Comment by lukashsmannheim** [comment]
Thank you very much! That solved my problem!
